### PR TITLE
Kotlin Alpha SDK spike

### PR DESCRIPTION
**Pull request by jkaster**
Snuck in some Swift fixes here too, because they were similar issues.

Includes the Kotlin RTL, generator, tests, and readme

### CODE REVIEW

File: kotlin/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+When the SDK is installed and the server location and API credentials are configured in your `looker.ini` file, it's ready to be used.
+
+Verify authentication works and that API calls will succeed with code similar to the following:
+
+```kotlin
+val localIni = "/Users/Looker/sdk-codegen/looker.ini"
+val settings = ApiSettingsIniFile(localIni, "Looker")
+
+val client: HttpClient = HttpClient(Apache) {
```
Comment (brianmadden) [inline kotlin/README.md]: This is only necessary when validating against a Looker install with a self-signed cert. Do our customer instances have non-self-signed certs? Regardless, I think it's probably worth noting here that the `TrustSelfSignedStrategy` and `NoopHostnameVerifier` are dangerous and that they likely _should not_ end up in anything that is production level code.

File: kotlin/src/main/com/looker/rtl/Transport.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            }
+        } + addQueryParams(path, queryParams)
+    }
+
+    inline fun <reified T> request(
+            method: HttpMethod,
+            path: String,
+            queryParams: Values = mapOf(),
+            body: Any? = null,
+            noinline authenticator: Authenticator? = null): SDKResponse {
```
Comment (brianmadden) [inline kotlin/src/main/com/looker/rtl/Transport.kt]: Why did we change the default here from `Authenticator = ::defaultAuthenticator` to make it nullable? Is this due to the needing an `Optional` class to wrap nullable fields?

File: kotlin/src/test/KotlinExample.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.http.ssl.SSLContextBuilder
+import java.io.File
+import java.lang.Thread.sleep
+
+class KotlinExample {
+    val localConfig = "/Users/Looker/Documents/sdk_codegen/looker.ini"
+    val settings = ApiSettingsIniFile(localConfig, "Looker")
+
+    /**
+     * Note: This is only to work with self-signed certs... this should probably be removed before this is posted anywhere
```
Comment (brianmadden) [inline kotlin/src/test/KotlinExample.kt]: Probably worth changing this comment to specify that the pattern should not be used in production code instead. :D

File: kotlin/src/test/TestUserSession.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy
+import org.apache.http.ssl.SSLContextBuilder
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import org.junit.Test as test
+
+
+class TestUserSession {
+//    val localIni = "/Users/looker/Documents/sdk_codegen/looker.ini"
+    val localIni = "/Users/looker/sdk-codegen/looker.ini"
```
Comment (brianmadden) [inline kotlin/src/test/TestUserSession.kt]: TODO: We should figure out a more elegant way to do this for testing.

Comment (jkaster) [reply kotlin/src/test/TestUserSession.kt]: Yup. I've got an updated comment on that somewhere I'm surprised isn't showing up in this PR yet. Lots of cleanup remaining. May even get to it later today after this spike lands.

File: kotlin/src/test/KotlinExample.kt (hunk 3 above)
Comment (jkaster) [reply kotlin/src/test/KotlinExample.kt]: Changed

File: kotlin/src/main/com/looker/rtl/Transport.kt (hunk 2 above)
Comment (jkaster) [reply kotlin/src/main/com/looker/rtl/Transport.kt]: I've added a TODO

File: kotlin/README.md (hunk 1 above)
Comment (jkaster) [reply kotlin/README.md]: Thanks, replaced this code sample

**Pull request merged**